src/PhpSpec/Formatter/JsonFormatter.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Chris Kruining <chrise@keruining.eu>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+ 
+namespace PhpSpec\Formatter
+{
+    use PhpSpec\Event\ExampleEvent;
+    use PhpSpec\Event\SpecificationEvent;
+    use PhpSpec\Event\SuiteEvent;
+
+    final class JsonFormatter extends ConsoleFormatter
+    {
+        private $data = [];
+
+        private const STATUS_NAME = [
+            ExampleEvent::PASSED  => 'passed',
+            ExampleEvent::PENDING => 'pending',
+            ExampleEvent::SKIPPED => 'skipped',
+            ExampleEvent::FAILED  => 'failed',
+            ExampleEvent::BROKEN  => 'broken',
+        ];
+
+        public function beforeSpecification(SpecificationEvent $event)
+        {
+            $this->data[$event->getSpecification()->getTitle()] = [];
+        }
+
+        public function afterExample(ExampleEvent $event)
+        {
+            $specification = $event->getSpecification()->getTitle();
+            $example = $event->getTitle();
+
+            $this->data[$specification][$example] = [
+                'status' => static::STATUS_NAME[$event->getResult()],
```

Comment:
use `self::` to access private constants. Late static binding to access private things is not working (well, even though you cannot trigger the bug as inheritance is forbidden, but then *any* late static binding is useless anyway and only moves work from the compile time to the runtime)